src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -313,174 +357,183 @@ void func_80AF2BC0(EnRu2* this, PlayState* play) {
         if ((cue != NULL) && (cue->id == 3)) {
             Animation_Change(&this->skelAnime, animation, 1.0f, 0.0f, Animation_GetLastFrame(animation), ANIMMODE_ONCE,
                              0.0f);
-            this->action = 4;
+            this->action = ENRU2_RAISE_ARMS;
         }
     }
 }
 
-void func_80AF2C54(EnRu2* this, s32 arg1) {
-    if (arg1 != 0) {
-        this->action = 5;
+/* At the end of Ruto's arms-raising animation, cues the next action: spawning the
+   Water Medallion. */
+void EnRu2_HoldArmsUp(EnRu2* this, s32 shouldHold) {
+    if (shouldHold != 0) {
+        this->action = ENRU2_AWAIT_SPAWN_WATER_MEDALLION;
     }
 }
 
-void func_80AF2C68(EnRu2* this, PlayState* play) {
+/* Checks to see if the Water Medallion should spawn. */
+void EnRu2_CheckWaterMedallion(EnRu2* this, PlayState* play) {
     CsCmdActorCue* cue;
 
     if (play->csCtx.state != CS_STATE_IDLE) {
         cue = play->csCtx.actorCues[6];
 
         if ((cue != NULL) && (cue->id == 2)) {
-            this->action = 6;
-            func_80AF2A38(this, play);
+            this->action = ENRU2_FINISH_WATER_MEDALLION_CS;
+            EnRu2_SpawnWaterMedallion(this, play);
         }
     }
 }
 
-void func_80AF2CB4(EnRu2* this, PlayState* play) {
-    func_80AF2AB4(this, play);
+void EnRu2_Action00(EnRu2* this, PlayState* play) {
+    EnRu2_SetupWaterMedallionCutscene(this, play);
 }
 
-void func_80AF2CD4(EnRu2* this, PlayState* play) {
-    func_80AF2B44(this, play);
+void EnRu2_Action01(EnRu2* this, PlayState* play) {
+    EnRu2_CheckBlueWarp(this, play);
 }
 
-void func_80AF2CF4(EnRu2* this, PlayState* play) {
-    func_80AF2978(this, play);
+void EnRu2_Action02(EnRu2* this, PlayState* play) {
+    EnRu2_Rise(this, play);
     EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF2B94(this);
+    EnRu2_UpdateEyes(this);
+    EnRu2_EndRise(this);
 }
 
-void func_80AF2D2C(EnRu2* this, PlayState* play) {
-    func_80AF2744(this, play);
+void EnRu2_Action03(EnRu2* this, PlayState* play) {
+    EnRu2_UpdateBgCheckInfo(this, play);
     EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF2BC0(this, play);
+    EnRu2_UpdateEyes(this);
+    EnRu2_RaiseArms(this, play);
 }
 
-void func_80AF2D6C(EnRu2* this, PlayState* play) {
-    s32 something;
+void EnRu2_Action04(EnRu2* this, PlayState* play) {
+    s32 shouldHold;
 
-    func_80AF2744(this, play);
-    something = EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF2C54(this, something);
+    EnRu2_UpdateBgCheckInfo(this, play);
+    shouldHold = EnRu2_UpdateSkelAnime(this);
+    EnRu2_UpdateEyes(this);
```

Comment:
I think it probably makes more sense to name the return of `UpdateSkelAnime` as `animDone`, meaning the animation has finished playing. Same for `lookingDownLeft` in action 13